src/probnum/filtsmooth/_utils.py

```diff
@@ -0,0 +1,159 @@
+
+    Examples
+    --------
+
+    Create two car-tracking problems with similar parameters and disjoint locations.
+
+    >>> import probnum.problems.zoo.filtsmooth as filtsmooth_zoo
+    >>> prob1, info1 = filtsmooth_zoo.car_tracking(
+    ...     measurement_variance=2.0, timespan=(0.0, 10.0), step=0.5
+    ... )
+    >>> measmods1 = np.asarray([info1["measurement_model"]] * len(prob1.locations))
+    >>> print(prob1.locations)
+    [0.  0.5 1.  1.5 2.  2.5 3.  3.5 4.  4.5 5.  5.5 6.  6.5 7.  7.5 8.  8.5
+     9.  9.5]
+
+    >>> prob2, info2 = filtsmooth_zoo.car_tracking(
+    ...     measurement_variance=2.0, timespan=(0.25, 10.25), step=0.5
+    ... )
+    >>> measmods2 = np.asarray([info2["measurement_model"]] * len(prob2.locations))
+    >>> print(prob2.locations)
+    [0.25 0.75 1.25 1.75 2.25 2.75 3.25 3.75 4.25 4.75 5.25 5.75 6.25 6.75
+     7.25 7.75 8.25 8.75 9.25 9.75]
+
+    Merge them with merge_regression_problems
+
+    >>> new_prob, new_models = merge_regression_problems(
+    ...     (prob1, measmods1), (prob2, measmods2)
+    ... )
+    >>> print(new_prob.locations)
+    [0.   0.25 0.5  0.75 1.   1.25 1.5  1.75 2.   2.25 2.5  2.75 3.   3.25
+     3.5  3.75 4.   4.25 4.5  4.75 5.   5.25 5.5  5.75 6.   6.25 6.5  6.75
+     7.   7.25 7.5  7.75 8.   8.25 8.5  8.75 9.   9.25 9.5  9.75]
+
+    If you have more than two problems that you want to merge, do this with functools.reduce.
+
+    >>> import functools
+    >>> prob3, info3 = filtsmooth_zoo.car_tracking(
+    ...     measurement_variance=2.0, timespan=(0.35, 10.35), step=0.5
+    ... )
+    >>> measmods3 = np.asarray([info3["measurement_model"]] * len(prob3.locations))
+    >>> new_prob, new_models = functools.reduce(
+    ...     merge_regression_problems,
+    ...     ((prob1, measmods1), (prob2, measmods2), (prob3, measmods3)),
+    ... )
+    >>> print(new_prob.locations)
+    [0.   0.25 0.35 0.5  0.75 0.85 1.   1.25 1.35 1.5  1.75 1.85 2.   2.25
+     2.35 2.5  2.75 2.85 3.   3.25 3.35 3.5  3.75 3.85 4.   4.25 4.35 4.5
+     4.75 4.85 5.   5.25 5.35 5.5  5.75 5.85 6.   6.25 6.35 6.5  6.75 6.85
+     7.   7.25 7.35 7.5  7.75 7.85 8.   8.25 8.35 8.5  8.75 8.85 9.   9.25
+     9.35 9.5  9.75 9.85]
+    """
+
+    regression_problem1, measurement_models1 = problem_and_likelihood1
+    regression_problem2, measurement_models2 = problem_and_likelihood2
+
+    # Check input types of measurement models: we explicitly want numpy arrays.
+    # More abstract merging (e.g. of lists or generators) is not supported currently.
+    errormsg_type_measmods = (
+        f"The measurement models are expected to be of type `{np.ndarray}'"
+        f"but type `{type(measurement_models1)}' was received."
+    )
+    if not isinstance(measurement_models1, np.ndarray):
+        raise TypeError(errormsg_type_measmods)
+    if not isinstance(measurement_models2, np.ndarray):
+        raise TypeError(errormsg_type_measmods)
```

Comment:
```python
if not (isinstance(measurement_models1, np.ndarray) and isinstance(measurement_models2, np.ndarray)):
  raise ...
```
?

Reply:
Two reasons I slightly prefer the current one:
1. an **or** would be better than an **and** here, right? 
2. The line would be so long that black would make the whole thing into 3-4 lines anyway, so the space is the same for both versions, but the current one is a bit more compact. I find it easier to read if there are fewer linebreaks for single statements

What do you think?

Reply:
> 1. an or would be better than an and here, right?

For readability, yes. Otherwise it's equivalent.

> 2. 

Yep, that suggestion was just out of personal preference, to avoid if-branches if possible. But it makes no big difference, that's true :) 
We can leave it as is, if you prefer!

Reply:
I mostly share your sentiment as to avoiding if branches, but I am mostly against nested ifs. Here, one check after the other seems fine to me. I will leave it, if you dont mind ;)